applications/GeoMechanicsApplication/custom_utilities/extrapolation_utilities.cpp

```diff
@@ -0,0 +1,89 @@
+// KRATOS___
+//     //   ) )
+//    //         ___      ___
+//   //  ____  //___) ) //   ) )
+//  //    / / //       //   / /
+// ((____/ / ((____   ((___/ /  MECHANICS
+//
+//  License:         geo_mechanics_application/license.txt
+//
+//  Main authors:    Anne van de Graaf
+//                   Gennady Markelov
+//
+
+#include "custom_utilities/extrapolation_utilities.h"
+#include "custom_utilities/linear_nodal_extrapolator.h"
+#include "includes/node.h"
+
+namespace Kratos
+{
+
+Matrix ExtrapolationUtilities::CalculateExtrapolationMatrix(const Geometry<Node>& rGeometry,
+                                                            GeometryData::IntegrationMethod IntegrationMethod,
+                                                            size_t                   ElementId,
+                                                            const NodalExtrapolator* pExtrapolator)
+{
+    KRATOS_TRY
+
+    LinearNodalExtrapolator default_extrapolator;
+    if (!pExtrapolator) pExtrapolator = &default_extrapolator;
+
+    const auto result = pExtrapolator->CalculateElementExtrapolationMatrix(rGeometry, IntegrationMethod);
+
+    KRATOS_ERROR_IF_NOT(result.size1() == rGeometry.size())
+        << "A number of extrapolation matrix rows " << result.size1() << " is not equal to a number of nodes "
+        << rGeometry.size() << " for element id " << ElementId << std::endl;
+
+    KRATOS_ERROR_IF_NOT(result.size2() == rGeometry.IntegrationPoints(IntegrationMethod).size())
+        << "A number of extrapolation matrix columns " << result.size2()
+        << " is not equal to a number of integration points "
+        << rGeometry.IntegrationPoints(IntegrationMethod).size() << " for element id " << ElementId
+        << std::endl;
+
+    return result;
+
+    KRATOS_CATCH("")
+}
+
+std::vector<std::optional<Vector>> ExtrapolationUtilities::CalculateNodalStresses(
+    const std::vector<std::size_t>& node_ids,
```

Comment:
This name doesn't follow the Kratos Style guide:
```suggestion
    const std::vector<std::size_t>& rNodeIds,
```

Reply:
fixed